src/compiler/checker.ts

```diff
@@ -4000,6 +4000,10 @@ namespace ts {
                 return createTypeReference((<TypeReference>type).target,
                     concatenate((<TypeReference>type).typeArguments, [thisArgument || (<TypeReference>type).target.thisType]));
             }
+            if (type.flags & TypeFlags.Tuple) {
+                resolveTupleTypeMembers(type as TupleType, thisArgument);

```

Comment:
How do I create a new tuple type safely? Can I just say `createNewTupleType(type.elementTypes)`?

And once I have the tuple type, I think I'll need to store `thisType` in a separate property so that `resolveTupleTypeMembers` can use it later, right? (I guess I could still immediately call `resultTupleTypeMembers` like I do in the current commit.)